src/server/pps/server/api_server.go

```diff
@@ -2921,6 +2965,15 @@ func (a *apiServer) inspectPipelineInTransaction(txnCtx *txnenv.TransactionConte
 		pipelineInfo.WorkersAvailable = int64(len(workerStatus))
 		pipelineInfo.WorkersRequested = int64(pipelinePtr.Parallelism)
 	}
+	unclaimedTasks, err := work.NewWorker(
+		a.env.GetEtcdClient(),
+		a.etcdPrefix,
+		driver.WorkNamespace(pipelineInfo),
+	).UnclaimedTasks(txnCtx.ClientContext)
```

Comment:
Just curious, are there other places where we instantiate a Worker instance in pachd? I'm trying to think of weird states that this struct might get into—probably it's fine to leave it, but part of me wonders if it's worth handling e.g. `Not Found` in case this races with `UpdatePipeline`? (is `NotFound` even the error you'd get? Maybe it makes sense to just leave this alond too)

Reply:
So I don't think you can get `not found` because `UnclaimedTasks` is implemented as taking the difference between two count operations in etcd and I don't think those can ever return `not found` they just return 0 if they don't find anything.